services/user.js

```diff
@@ -200,14 +201,22 @@ class UserService {
 
     async getUser(params, context) {
         verifySession(context)
-            .verifyInitialized()
-            .verifyPermission(USER_PERMISSION_CONSTANTS.ADMIN.MANAGE_USER);
-        
+            .verifyInitialized();
+        await this.#isPermitted(context?.userInfo, USER_PERMISSION_CONSTANTS.ADMIN.MANAGE_USER);
+
         if (!params?.userID) {
             throw new Error(SUBMODULE_ERROR.INVALID_USERID);
         }
+        const isFederalLeadOnly = await this.#isManageUserScope(context?.userInfo, ROLES.FEDERAL_LEAD);
+        if (context?.userInfo?.role === ROLES.FEDERAL_LEAD && !isFederalLeadOnly) {
+            throw new Error(ERROR.INVALID_FEDERAL_LEAD_REQUEST);
+        }
+
+        const filters = {
+            _id: params.userID,
+            ...(isFederalLeadOnly ? { role: ROLES.FEDERAL_LEAD } : {})
```

Comment:
Here, if the scope we got is "role" then use the scope value in the filters. If the scope is "all" then no filter is needed.

Reply:
Yes, the specific role is not removed.